ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -106,6 +110,16 @@ public UnresolvedPlan visitSearchFilterFrom(SearchFilterFromContext ctx) {
         visit(ctx.fromClause()));
   }
 
+  /**
+   * Describe command.
+   */
+  @Override
+  public UnresolvedPlan visitDescribeCommand(DescribeCommandContext ctx) {
```

Comment:
Could you please paste the result of explain command, so that it is easier to understand what you are trying to do here.

Reply:
Could you please comment the logic over here since it is slightly different from rest of the commands.

Reply:
I borrow code from the describe implementation for SQL:
https://github.com/opensearch-project/sql/blob/7cbb121cb3ad144a5e6d5b9acb9ae8e6da01688e/sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java#L73-L77

The syntax for SQL and PPL are different, hence we have different parsing logic. However, the overall idea is the same. At its core, the logic is as follows (simplified line by line):
```java
// get index name
final String tableName = ctx.tableSource().toString();
// mappingTable gets the metadata table (_ODFE_SYS_TABLE_MAPPINGS) for the index
final Relation table = new Relation(mappingTable(tableName));
// this gets all rows in the metadata table, while each row corresponds to a field in the queried index
return new Project(Collections.singletonList(AllFields.of())).attach(table);
```
The additional `internalVisitExpression` and `qualifiedName` is for parsing [complex identifiers](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/general/identifiers.rst)

I'll add some comments to this function.

Explain:
```
$ curl -XPOST https://localhost:9200/_plugins/_ppl/_explain -u 'admin:admin' -k -H 'Content-Type: application/json' -d '{"query": "describe opensearch_dashboards_sample_data_flights"}'

{
  "root": {
    "name": "ProjectOperator",
    "description": {
      "fields": "[TABLE_CAT, TABLE_SCHEM, TABLE_NAME, COLUMN_NAME, DATA_TYPE, TYPE_NAME, COLUMN_SIZE, BUFFER_LENGTH, DECIMAL_DIGITS, NUM_PREC_RADIX, NULLABLE, REMARKS, COLUMN_DEF, SQL_DATA_TYPE, SQL_DATETIME_SUB, CHAR_OCTET_LENGTH, ORDINAL_POSITION, IS_NULLABLE, SCOPE_CATALOG, SCOPE_SCHEMA, SCOPE_TABLE, SOURCE_DATA_TYPE, IS_AUTOINCREMENT, IS_GENERATEDCOLUMN]"
    },
    "children": [
      {
        "name": "OpenSearchSystemIndexScan",
        "description": {
          "request": "OpenSearchDescribeIndexRequest{indexName='opensearch_dashboards_sample_data_flights'}"
        },
        "children": []
      }
    ]
  }
}
```

Reply:
closing because no longer applies